lib/fixture/locator/helpers.ts

```diff
@@ -42,13 +50,61 @@ const buildTestingLibraryScript = async ({config}: {config: Config}) => {
 const synchronousQueryNames = allQueryNames.filter(isNotFindQuery)
 
 const queriesFor = (pageOrLocator: Page | Locator) =>
-  synchronousQueryNames.reduce(
-    (rest, query) => ({
+  allQueryNames.reduce((rest, query) => {
+    if (isFindQuery(query)) {
+      return {
+        ...rest,
+        [query]: async (
+          ...[id, options, waitForElementOptions]: Parameters<Queries[FindQuery]>
+        ) => {
+          const synchronousOptions = ([id, options] as const).filter(Boolean)
+
+          const locator = pageOrLocator.locator(
+            `${queryToSelector(findQueryToQueryQuery(query))}=${JSON.stringify(
+              synchronousOptions,
+              replacer,
+            )}`,
+          )
+
+          try {
+            // TODO: fall back to `asyncUtilTimeout` here
+            // TODO: support passing `state` option through?
+            await locator.first().waitFor({timeout: waitForElementOptions?.timeout})
+          } catch (error) {
+            // In the case of a `waitFor` timeout from Playwright, we want to
+            // surface the appropriate error from Testing Library, so run the
+            // query one more time as `get*` knowing that it will fail with the
+            // error that we want the user to see instead of the `TimeoutError`
+            if (error instanceof errors.TimeoutError) {
+              return (
+                pageOrLocator
+                  .locator(
+                    `${queryToSelector(findQueryToGetQuery(query))}=${JSON.stringify(
+                      synchronousOptions,
+                      replacer,
+                    )}`,
+                  )
+                  // TODO: is this the best way to 'dereference' the `Locator`?
```

Comment:
I think so. For a moment, I considered [`Locator#elementHandles()`](https://playwright.dev/docs/api/class-locator#locator-element-handles), however, since we are trying to percolate the same error as was found above when using `.waitFor` and we're planning on sending through a `state` property I reckon we need to do exactly the same thing here.

FYI, I think you'll need to use `.first()` here, too?

Reply:
(for the record it seems the query error is thrown before the strict error so this isn't fully necessary but I think it's the right move anyways in case anything changes subtly in either library)